internal/template/character/stam.go

```diff
@@ -102,18 +111,51 @@ func (c *Character) DashLength() int {
 	}
 }
 
-func (c *Character) Jump(p map[string]int) (action.Info, error) {
-	f := 30
+func (c *Character) DashToJumpLength() int {
 	switch c.CharBody {
-	case info.BodyBoy, info.BodyGirl:
-		f = 31
-	case info.BodyMale:
-		f = 28
-	case info.BodyLady:
-		f = 32
-	case info.BodyLoli:
-		f = 29
+	case info.BodyGirl, info.BodyLoli:
+		return 4
+	case info.BodyMale, info.BodyLady:
+		return 3
+	case info.BodyBoy:
+		return 2
+	default:
+		return 3
 	}
+}
+
+func (c *Character) Jump(p map[string]int) (action.Info, error) {
+	if c.StatusIsActive(player.XianyunAirborneBuff) {
+		c.Core.Player.SetAirborne(player.AirborneXianyun)
+		// 4/8 for claymore/bow/catalyst and 5/9 for sword/polearm
+		lowPlunge := 4
+		switch c.Weapon.Class {
+		case info.WeaponClassSword, info.WeaponClassSpear:
+			lowPlunge = 5
+		}
+
+		highPlunge := 8
+		switch c.Weapon.Class {
+		case info.WeaponClassSword, info.WeaponClassSpear:
+			highPlunge = 9
+		}
+		return action.Info{
+			Frames: func(a action.Action) int {
+				switch a {
+				case action.ActionLowPlunge:
+					return lowPlunge
+				case action.ActionHighPlunge:
+					return highPlunge
+				default:
+					return 60
```

Comment:
Please add a comment that you expect this case to be ineffectual because sim will error out on xianyun buffed jump into anything else than low plunge/high plunge.